search/src/main/java/edu/artic/search/SearchBaseViewModel.kt

```diff
@@ -174,4 +176,36 @@ open class SearchBaseViewModel @Inject constructor(
     open fun onClickSeeAll(header: Header) {
 
     }
+
+    /**
+     * Returns suggested artworks.
+     */
+    protected fun getSuggestedArtworks(searchSuggestionsDao: ArticSearchObjectDao,
+                                       objectDao: ArticObjectDao
+    ): Observable<MutableList<ArticObject>> {
+        return searchSuggestionsDao.getDataObject()
+                .toObservable()
+                .map { suggestedSearchOptions -> suggestedSearchOptions.searchObjects }
+                .flatMap { idsList ->
+                    /**
+                     * Database does not preserve the order of ids.
+                     */
+                    objectDao.getObjectsByIdList(idsList)
+                            .toObservable()
+                            .map { unSortedObjects ->
+                                /**
+                                 * Sorting the objects based on idsList
+                                 */
+                                val sortedObjects = mutableListOf<ArticObject>()
+                                for (id in idsList) {
```

Comment:
I think this would look better with the Kotlin Stream operators, as

    val sortedObjects: List<ArticObject> = idsList.mapNotNull { id ->
        unSortedObjects.find { it.nid == id }
    }

or even

    return@map idsList.mapNotNull { id ->
        unSortedObjects.find { it.nid == id }
    }

Then you can make the function return an `Observable<List<ArticObject>>`. It'll match the type of `suggestedArtworks` too that way. Iirc `mapNotNull` preserves order the same way `for...in` does.